common/client-libs/gateway-client/src/client/mod.rs

```diff
@@ -221,7 +217,7 @@ impl<C, St> GatewayClient<C, St> {
 
     #[cfg(target_arch = "wasm32")]
     pub async fn establish_connection(&mut self) -> Result<(), GatewayClientError> {
-        let ws_stream = match JSWebsocket::new(&self.gateway_address) {
+        let ws_stream = match JSWebsocket::new(&self.gateway_addresses.primary.as_ref()) {
```

Comment:
what about the fallback here?

Reply:
IIRC wasm client requires WSS and that means no IP fallback possible, because we need a hostname for TLS